### PR TITLE
Feature 556 report articles button

### PR DESCRIPTION
**Pull request by John-Paul-Larkin**
# ✨ Codu Pull Request 💻

Codu Logo

## Pull Request details:
This PR adds a report article option to the side menu.
It reuses the the reportComment component, but creates a specific email template for articles(rather than comments).

There is an issue with the implementation, so this is not ready to be merged yet. 
The way I have it now, the ReportComment component is nested within the headlessui Menu component. This interfers with the tab index of the modal. If the user presses space it will close the modal.

### CODE REVIEW

File: components/ArticleMenu/ArticleMenu.tsx [hunk 1]
```diff
@@ -188,6 +189,13 @@ const ArticleMenu = ({
                     </a>
                   </Menu.Item>
                 ))}
+                <ReportComments
+                  postTitle={postTitle}
+                  name={postUsername}
+                  postId={postId}
+                  postUsername={postUsername}
+                  postUrl={postUrl}
+                />
```
Comment (John-Paul-Larkin) [inline components/ArticleMenu/ArticleMenu.tsx]: This is the issue. We could possibly put the component as a button below the bookmark?
Something like this.
Screenshot from 2023-10-17 20-51-17
Maybe that is not the look you want, so im open to suggestions.

Comment (NiallJoeMaher) [reply components/ArticleMenu/ArticleMenu.tsx]: I was thinking of adding it as an option to the 3 dots menu: 
Screenshot 2023-10-18 at 07 50 20

**Review by NiallJoeMaher — CHANGES_REQUESTED** [review]

**Comment by John-Paul-Larkin** [comment]
Hi Niall. That is how I have implemeted it in this PR.
The issue is that i have the reportComments component within the headlessui Menu component. This interfers with the tab index, or the focus of the dialog(modal). 

While typing this I just thought of react portal. That could be a solution to this problem. I will try it now.

**Comment by NiallJoeMaher** [comment]
> Hi Niall. That is how I have implemeted it in this PR. The issue is that i have the reportComments component within the headlessui Menu component. This interfers with the tab index, or the focus of the dialog(modal).
> 
> While typing this I just thought of react portal. That could be a solution to this problem. I will try it now.

Before you go too far, we have HeadlessUI installed and I think they have this solved with portals by default: 

https://headlessui.com/react/dialog

Will make things accessible too! 

I’ll open up an issue when it’s finished too to make them work in both light and dark mode (no need to tackle that in this issue).

**Comment by John-Paul-Larkin** [comment]
Im new to headlessUI and i think it is awesome. The docs are great too. I will read over them and redo the modal component.

File: components/ArticleMenu/ArticleMenu.tsx [hunk 2]
```diff
@@ -174,23 +162,42 @@ const ArticleMenu = ({
...
-                    >
-                      {item.label}
-                    </a>
-                  </Menu.Item>
-                ))}
-              </Menu.Items>
+              {/* <Popover.Panel className="absolute z-10"> */}
+              <Popover.Panel className="origin-top-right absolute bottom-14 right-0 lg:left-16 lg:bottom-0 mt-2 w-48 rounded-md shadow-lg py-1 bg-white dark:bg-white ring-1 px-1 ring-black ring-opacity-5 focus:outline-none">
+                <div
+                // className="block px-4 py-2 text-neutral-900 dark:text-neutral-700 hover:bg-neutral-200 rounded"
```
Comment (NiallJoeMaher) [inline components/ArticleMenu/ArticleMenu.tsx]: I'm assuming this comment isn't needed?

File: components/Comments/ReportComments.tsx [hunk 3]
```diff
@@ -27,137 +33,187 @@ export const ReportComments = (props: Props) => {
...
-        </dialog>
-      )}
+          </Dialog.Panel>
+        </div>
+      </Dialog>
     </>
   );
 };
+
+//  {!postTitle && (
```
Comment (NiallJoeMaher) [inline components/Comments/ReportComments.tsx]: You can remove the unused code. 🦾

**Review by NiallJoeMaher — CHANGES_REQUESTED** [review]

**Comment by John-Paul-Larkin** [comment]
I have redone this. I will close this and do a new PR tomorrow.

**Comment by John-Paul-Larkin** [comment]
Ah. I pushed a commit without realising it would ask you for review. I'm not quite finished with this and will do amother pr later or tomorrow. Thanks Niall

**Pull request merged**